### ISSUE TITLE
Image push fails when tags contains 2 or more items

### ISSUE THREAD

**Issue by zetaab** [docker/build-push-action#126]
#### Steps to reproduce this issue

use following workflow https://github.com/ElisaOyj/kops-autoscaler-openstack/blob/master/.github/workflows/master.yml

1. it should have login to docker registry
2. it should have push to registry with more than 2 tags

#### Expected behaviour

I expect that credentials are used to all tags that are listed. 

#### Actual behaviour

https://github.com/ElisaOyj/kops-autoscaler-openstack/runs/1122690450?check_suite_focus=true I have executed this now twice but always only the first tag will use credentials and other tags will return http code 401 from registry.

We can see only one tag now in registry `latest` but another one is missing

**Comment by crazy-max** [comment]
@zetaab Encountered the [same issue](https://github.com/crazy-max/artifactory-cleanup/runs/1125892117?check_suite_focus=true#step:8:2098) with GitHub Container Registry but not with DockerHub:

```
#92 exporting to image
#92 exporting layers
#92 exporting layers 4.1s done
#92 exporting manifest sha256:ef6889dcdcfebe634668dec3bfd438961ccca2547e45e155fbd122584418b951 0.0s done
#92 exporting config sha256:f07ac3e9d4c0e93f32473f36dc415d0feb4322b52ef7b7d25e31647d66516498 done
#92 exporting manifest sha256:b8d4d85dfd2e1a4ce254c324315e8f2c4bb92794f0f16d74e55e9f65e41d03f2 done
#92 exporting config sha256:73435118f72c21eeb48d58d2e733468feba6be36789e544cea01256581cf69b2
#92 exporting config sha256:73435118f72c21eeb48d58d2e733468feba6be36789e544cea01256581cf69b2 done
#92 exporting manifest sha256:718979d6ff040b74f7fd56e92baeb4878bc0c0103423d652c5a817dd13f5f463 done
#92 exporting config sha256:f5465fad284a2a1849958c9abf792d02eceeaeaa30be600f742df0d8482eb920 done
#92 exporting manifest sha256:23fc1a7c23f3b728b25a1f3d1a9e354d9e3e9ed7d0f1bced957c306731c862a4 done
#92 exporting config sha256:61e8aa7c5b3b146d2336c9369d49fbde6daea256a6a36293da22537085fa2386 done
#92 exporting manifest sha256:0fd53213d45b9d57a7dc370b7de3f7ecc46c3bf64cbe45b6bea02cb782fab8ae done
#92 exporting config sha256:f4f66070ab7dc523451ba81da86111830df289d02e93f06b4cb846decbbdb18d done
#92 exporting manifest sha256:b168ee11952b9fd0144f4ddd2d0e8678a92daf839d734134ee5fbe706e25f915 done
#92 exporting config sha256:6e272b05dccf55753a581b0509e80d68c81598419a163e349d10d794173f0d80 done
#92 exporting manifest list sha256:d99df961bd7dfb5fc3aab773b132632edabad08ecd5f8b1526e9b9a2a6d6f39a done
#92 pushing layers
#92 pushing layers 1.9s done
#92 pushing manifest for docker.io/***/artifactory-cleanup:0.1.0
#92 pushing manifest for docker.io/***/artifactory-cleanup:0.1.0 2.0s done
#92 pushing layers 1.4s done
#92 pushing manifest for ghcr.io/crazy-max/artifactory-cleanup:0.1.0
#92 pushing manifest for ghcr.io/crazy-max/artifactory-cleanup:0.1.0 4.9s done
#92 pushing layers 1.5s done
#92 pushing manifest for docker.io/***/artifactory-cleanup:latest
#92 pushing manifest for docker.io/***/artifactory-cleanup:latest 1.7s done
#92 pushing layers 0.2s done
#92 pushing manifest for ghcr.io/crazy-max/artifactory-cleanup:latest 0.1s done
#92 ERROR: failed commit on ref "manifest-sha256:b8d4d85dfd2e1a4ce254c324315e8f2c4bb92794f0f16d74e55e9f65e41d03f2": unexpected status: 401 Unauthorized
```

So maybe an issue with the registry. WDYT @tonistiigi?

**Comment by chriskinsman** [comment]
driver-opts worked for me

**Comment by chriskinsman** [comment]
driver-opts didn't work here.  Still when pushing multiple tags:

https://github.com/chriskinsman/tilloo/runs/1147251118?check_suite_focus=true

**Comment by crazy-max** [comment]
@zetaab @chriskinsman 

It's now working for me (https://github.com/crazy-max/test-docker-action/runs/1150312201?check_suite_focus=true#step:7:228)
For support about GHCR see https://github.com/docker/build-push-action/issues/127#issuecomment-696409205.

**Comment by chriskinsman** [comment]
@crazy-max Tried to make my build workflow as similar as possible to yours.  Couldn't get it working until I removed the cache action.  Now without caching it works.  Once I changed the cache key/restore-keys to a new key so I no longer used any old cached output it worked.

**Comment by crazy-max** [comment]
@chriskinsman 

> Couldn't get it working until I removed the cache action. Now without caching it works.

Hum that's strange, the cache is not propagated if a step fails:

![image](https://user-images.githubusercontent.com/1951866/93915648-da6e3a00-fcf7-11ea-8d81-eda0b3ebd292.png)

I will dig this out. Thanks for your feedback.

**Comment by zetaab** [comment]
still not working for me

**Comment by djlauk** [comment]
I am having what looks like the same issue. I'm using GitHub Container Registry (ghcr.io), and one of the tags gets pushed, the other doesn't.

**Comment by crazy-max** [comment]
@zetaab @djlauk Should be fixed now (docker/setup-buildx-action#27)

**Comment by djlauk** [comment]
Sorry for the delayed response. I can confirm it works now for me.
Thanks a lot!
